account_asset_management/models/account_asset.py

```diff
@@ -735,12 +738,17 @@ def compute_depreciation_board(self):
                 )
                 residual_amount = asset.depreciation_base - depreciated_value
                 amount_diff = currency.round(residual_amount_table - residual_amount)
+                amount_diff = amount_diff / (
+                    (total_table_lines - len(move_check_lines)) or 1
+                )
                 if amount_diff:
                     # We will auto-create a new line because the number of lines in
                     # the tables are the same as the posted depreciations and there
                     # is still a residual value. Only in this case we will need to
                     # add a new line to the table with the amount of the difference.
                     if len(move_check_lines) == total_table_lines:
+                        if asset.method_period == "year":
```

Comment:
Why this adjustment? It should be documented in the code itself.

Reply:
In periods set to years it has been found that table_i_start is not set correctly. This is an error that is currently occurring. You can check this.

although this fixes the problem, I don't see the possibility to correct this when assigning the table_i_start value.

1. Create asset
![Selección_773](https://github.com/OCA/account-financial-tools/assets/6359121/29165156-a59b-4eee-a4d3-87b10bd0dbb4)

1. Create first four move
![Selección_774](https://github.com/OCA/account-financial-tools/assets/6359121/20d808cd-dbaa-4bcf-b30b-9fa70cfd98fa)

1. Set to draft and change Number of Years to 4
![Selección_775](https://github.com/OCA/account-financial-tools/assets/6359121/8e30cc16-630d-48e3-981f-aa2c13b9adc5)

1. Set to draft and change Number of Years to 4
![Selección_776](https://github.com/OCA/account-financial-tools/assets/6359121/d7cfa68e-36d9-4f46-abbd-5d3faa8aaaea)

![Selección_777](https://github.com/OCA/account-financial-tools/assets/6359121/1119e7dc-3415-4bb5-8e95-988dc5f1a9ec)

Reply:
ping @pedrobaeza